File: medium/134. Gas Station/134. Gas Station.py
from typing import List


class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        l = len(gas)  # Gas length
        tank = 0
        curr_tank = 0
        start = 0

        for i in range(l):
            tank += gas[i] - cost[i]
            curr_tank += gas[i] - cost[i]

            if curr_tank < 0:
                curr_tank = 0
                start = i + 1

        return start if tank >= 0 else -1


solution = Solution()

print(solution.canCompleteCircuit(gas=[1, 2, 3, 4, 5], cost=[3, 4, 5, 1, 2]))
print(solution.canCompleteCircuit(gas=[2, 3, 4], cost=[3, 4, 3]))

File: medium/134. Gas Station/134. Gas Station.ts
function canCompleteCircuit(gas: number[], cost: number[]): number {
    const l = gas.length
    let tank = 0, curr_tank = 0
    let start = 0

    for (let i = 0; i < l; i++) {
        tank += gas[i] - cost[i]
        curr_tank += gas[i] - cost[i]

        if (curr_tank < 0) {
            curr_tank = 0
            start = i + 1
        }
    }

    return tank >= 0 ? start : -1
};